pkg/compression/compression.go

```diff
@@ -43,24 +78,44 @@ var compressionAlgos = map[string]struct {
 	"gzip":  {[]byte{0x1F, 0x8B, 0x08}, GzipDecompressor},                 // gzip (RFC 1952)
 	"bzip2": {[]byte{0x42, 0x5A, 0x68}, Bzip2Decompressor},                // bzip2 (decompress.c:BZ2_decompress)
 	"xz":    {[]byte{0xFD, 0x37, 0x7A, 0x58, 0x5A, 0x00}, XzDecompressor}, // xz (/usr/share/doc/xz/xz-file-format.txt)
+	"zstd":  {[]byte{0x28, 0xb5, 0x2f, 0xfd}, ZstdDecompressor},           // zstd (http://www.zstd.net)
+}
+
+// compressors maps an algorithm to its compression function
+var compressors = map[string]CompressorFunc{
+	"gzip":  GzipCompressor,
+	"bzip2": Bzip2Compressor,
+	"xz":    XzCompressor,
+	"zstd":  ZstdCompressor,
```

Comment:
If the callers need to deal with the strings, please turn them into a typed Go enum.

… and then `DecompressorFunc` and `CompressorFunc` can be removed, and become methods on that typed enum.  (Well, `DecompressorFunc` needs to continue to exist so that we don’t break the API.)

Reply:
are you fine with the last version where I changed the function to look like:

```
func GetCompressor(dest io.Writer, name string, level int) (io.WriteCloser, error)
```

so now we don't expose `CompressorFunc` at all.  Do you still want to have a Go enum for the compression algorithms?

Reply:
Yes, please

- let’s not have an untyped
  ```go
  desiredCompressionFormat = "gzip"
  ```
  when it could use a `compression.Gzip` constant
- A named Go type would prevent accidentally/unnecessarily mixing raw strings and format values.
- Depending on what the type exactly ends up being, making `GetCompressor` a method on the format type might eliminate the
  ```go
  c, found := compressors[name]
		if !found {
  ```
  failure path.  (If not, no harm done.)
- OTOH, that might force us to add a `compression.GetFormatFromString()`, to set types.SystemContext.CompressionFormat` with the typed value — but that would be a good thing, because invalid values would have to be recognized early at the CLI level, instead of deep inside the `compressGoroutine`.